history-network.md

```diff
@@ -0,0 +1,300 @@
+# Chain History Storage Network
+
+This document is a preliminary specification for a networking protocol that supports on-demand availability of Ethereum chain history data.
+
+## Overview
+
+Chain history data consists of historical block headers and block bodies, where a block body consists of transactions and transaction receipts.
+
+The data stored in the chain history storage network will support the following [eth](https://github.com/ethereum/devp2p/blob/master/caps/eth.md) protocol requests:
+
+* `GetBlockHeaders (0x03)`
+* `GetBlockBodies (0x05)`
+* `GetReceipts (0x0f)`
+
+The chain history storage network is a [Kademlia](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) DHT that forms an overlay network on top of a [Discovery v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md) network.
+
+The history network uses the following protocol messages from the Discovery v5 network:
+
+* `PING (0x01)`
+* `PONG (0x02)`
+* `FINDNODE (0x03)`
+* `NODES (0x04)`
+* `TALKREQ (0x05)`
+* `TALKRESP (0x06)`
+
+The `TALKREQ` and `TALKRESP` protocol messages are application-level messages whose contents are specific to the history protocol. We specify these messages below.
+
+The history protocol uses the node table structure from the Discovery v5 network and the lookup algorithm from section 2.3 of the Kademlia paper.
+
+### Data
+
+Types:
+
+* Block headers
+* Block bodies
+    * Transactions
+    * Receipts
+
+Lookups:
+
+* Block header by block header hash
+* Block body by block header hash
+* Block receipts by block header hash
+
+This specification does not support block lookups by number or transaction lookups by hash. To support these lookups, we will require a specification for a block header accumulator so that we can return the canonical block for a given height or the transaction for a given hash included in some canonical block.
+
+### Data Bridge
+
+Participants in the `eth` protocol are responsible for injecting data into the network. These nodes act as a bridge between the `eth` protocol and the chain history network. Once the data is present within the network, that data is distributed to nodes "close" to that data. We define a notion of distance between some node and some data item below.
```

Comment:
This also isn't quite accurate, since most participants of the DevP2P `eth` protocol will likely be unaware of the portal network.  I believe a more accurate way of stating this would be something like:

> There will be custom pieces of software which are responsible for extracting data from the DevP2p `eth` protocol and seeding it into the portal network.  We use the term "bridge node" to refer to a node performing these duties.